Added typescript definitions file

goooseman opened the pull request:
Adds `index.d.ts` with Typescript typings for the library.

`index.d.ts` is the default definitions locations. If the file is in the project root and is named `index.d.ts`, then there is no need to modify `package.json`.

Janealter commented:
@goooseman Are you sure the types are correct? I'm confused about the `new` keyword. An instance of `ArrayType` must not be defined via `new`.

goooseman commented:
@Janealter ok, my bad: I've just copied those types from `ref-array` typing without proper inspection.

I've removed the `new` function from the default export.
Checked out the tests to be sure that those typings are valid at the moment.
And refactored those typing a little to eliminate the duplicated interface.

P.S. I will update Janealter/ref-struct-napi#1 after this MR is merged.

Janealter commented on index.d.ts:
```diff
@@ -0,0 +1,50 @@
+   */
+  untilZeros(buffer: Buffer): ArrayTypeInstance<T>;
+
+  new (length?: number): ArrayTypeInstance<T>;
+  new (data: number[], length?: number): ArrayTypeInstance<T>;
+  new (data: Buffer, length?: number): ArrayTypeInstance<T>;
```
I think this is also not necessary

Janealter replied:
Oh, sorry, you're right)
